### ISSUE TITLE
Suporte para Kodi 19 Matrix

### ISSUE THREAD

**Issue by marcusmuller** [olavopeixoto/plugin.video.brplay#134]
Kodi usa agora o Python 3

**Comment by marcusmuller** [comment]
O addon não pode ser aberto nesta nova versão.

**Comment by lucaskercher** [comment]
Aqui as imagens mostrando os erros no Kodi 19 (Matrix):
![erro](https://user-images.githubusercontent.com/79429124/108951210-56bdce00-7646-11eb-890b-f43bf1fd6f6f.png)
![erro 2](https://user-images.githubusercontent.com/79429124/108951242-5fae9f80-7646-11eb-85b7-b84e3e146a64.png)

**Comment by olavopeixoto** [comment]
Eu já fiz bastante modificações pra compatibilidade com o Matrix mas tem muita coisa pra verificar e algumas dependências em bibliotecas pra resolver.
O mais complicado é manter 2 versões no mesmo código, por isso ainda não lancei nada.
Ainda estou avaliando se é melhor manter 2 ao mesmo tempo ou parar completamente o desenvolvimento do Leia e focar no Matrix. Acho que será melhor assim mas ainda acho cedo pra mudar.

**Comment by marcusmuller** [comment]
Concordo também em aguardar para mais estabilidade do Matrix. Vejo rewies em tá surgindo muita incompatibilidade com drives em PC, Android e NVidia Shield.

**Comment by olavopeixoto** [comment]
Se quiser testar ja tem um branch com código pro matrix. O proxy local ainda não esta funcionando significa que mídias em dash (.mpd) do globoplay e globosat não irão funcionar. Mas do resto acho que esta ok.

https://github.com/olavopeixoto/plugin.video.brplay/tree/Matrix

**Comment by eder-erl** [comment]
Ao testar o código em um Smartphone a listagem de canais na versão otimizada para o Kodi Matrix informa apenas o provedor de origem do streaming conforme o print abaixo, todavia todos canais abrem e exibem os conteúdos normalmente.

![Screenshot_20210309-102122_Kodi](https://user-images.githubusercontent.com/80330028/110477753-4f74d680-80c2-11eb-9c54-749b04901780.jpg)

**Comment by olavopeixoto** [comment]
O Matrix está exibindo informações diferentes do Leia mas se você mudar a ordenação existem critérios que fica com a exibição habitual

**Comment by dudumaroja** [comment]
minha sugestão seria.. parar a leia. esta funcionando muito bem como esta e so manter em caso de bugs.. e começar a focar no matrix que vai ser o futuro... creio que o Peyton 3 por exemplo e uma mudança que não terá volta.. não sei como funciona o sistema de repositório.. mas o ideal seria através do repositório cada sistema pega a versão que convém.

**Comment by leonelsr** [comment]
OSMC agora atualizou para 19.1 Matrix. BR Play ficou quebrado porque a versão 2.0.1 tá dando aquele "Invalid Token".

Alguém tem alguma pista do que precisa ser portado do master pra lá, pra voltar a funcionar?

**Comment by xicobiu** [comment]
> OSMC agora atualizou para 19.1 Matrix. BR Play ficou quebrado porque a versão 2.0.1 tá dando aquele "Invalid Token".
> 
> Alguém tem alguma pista do que precisa ser portado do master pra lá, pra voltar a funcionar?

Matrix ainda não 100% compatível. Você deve usar a versão 18 Leia.

Esse erro "Invalid Token" está sendo discutido aqui #158

**Comment by leonelsr** [comment]
> > OSMC agora atualizou para 19.1 Matrix. BR Play ficou quebrado porque a versão 2.0.1 tá dando aquele "Invalid Token".
> > Alguém tem alguma pista do que precisa ser portado do master pra lá, pra voltar a funcionar?
> 
> Matrix ainda não é 100% compatível. Você deve usar a versão 18 Leia.
> 
> Esse erro "Invalid Token" está sendo discutido aqui #158

Sim.

No update de [8 de agosto](https://osmc.tv/2021/08/osmcs-august-update-is-here-with-kodi-v19/) o **OSMC** puxou o Kodi 19.1 (Matrix) para o status de "stable" deles (ou seja, não vai mais ter esforços de atualizações para o Kodi 18 no OSMC). **LibreELEC** fez a mesma coisa [anteontem](https://libreelec.tv/2021/08/26/libreelec-matrix-10-0/)!!!

Boxes Android recentes todas já suportam o Matrix também.
A tendência é cada vez mais gente usar o Matrix, e cada vez menos desenvolvedores de addons suportarem o Leia.

Estou me voluntariando pra ajudar no trabalho de portar o BR Play para Python 3! Bora? Vou dar uma olhada nos commits mais recentes da v1 pra ver o que falta levar pra v2...

**Comment by crisclacerda** [comment]
Alguma novidade no suporte para o matrix?

**Comment by olavopeixoto** [comment]
O plugin está funcional no matrix, existe um branch específico para o matrix onde você pode fazer o download, existe também o repositório brplay com ambas as versões disponíveis.
A única pendência funcional até o momento é o proxy http interno do plugin, se você não precisa dessa funcionalidade, ele está funcionando normalmente.
